lib/message.rb

```diff
@@ -71,8 +71,31 @@ def as_json(options = {})
       hash
     end
 
+
+    attr_reader :events
+
+    def events?
+      !events.nil? && !events.empty?
+    end
+
+    def events=(events)
+      unless events.respond_to?(:map)
+        raise TypeError, "unable to iterate over #{events}, events must respond to #map"
+      end
+
+      @events = events.map do |event_data|
```

Comment:
Ah, I should have mentioned this – the event field is read-only so we don't need to be able to assign objects fields to it.

Reply:
Ah! good to know :)